src/projectile.py

```diff
@@ -98,11 +97,13 @@ class Beam(Projectile):
 
     beam_range: float
 
-    def setup(self):
-        self.beam_range = self.sprite.width
+    def setup(self, beam_range:float, explodes: bool = False):
+        self.beam_range = beam_range
+        self.explodes = explodes
```

Comment:
Sure I can envision a beam that causes explosions when it hits something. Why not